File: AndroidWithJavaAndXML/app/src/main/java/com/salmakhd/android/androidwithjavaandxml/intents/MainActivity.java
package com.salmakhd.android.androidwithjavaandxml.intents;

import android.os.Bundle;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;

import com.salmakhd.android.androidwithjavaandxml.R;
import com.salmakhd.android.androidwithjavaandxml.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {
    private TextView salmaSadText;
    private ImageView imageView;
    private Button button;
    private ActivityMainBinding binding;
    private TextView text;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(R.layout.activity_main);
        text = findViewById(R.id.myTextView);
        // get data from previous activity
        String email = getIntent().getStringExtra("email");
        text.setText(email);
//
//        button.setOnClickListener(new View.OnClickListener() {
//            @Override
//            public void onClick(View v) {
//                onButtonClicked(v, salmaSadText.getText().toString());
//            }
//        });
//        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
//            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
//            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
//            return insets;
//        });
//        imageView.setImageResource(R.drawable.ic_launcher_background);
//    }
//
//    private void onButtonClicked(View v, String textToBeDisplayed) {
//        Log.v("Button", "The button was just clicked: " + textToBeDisplayed);
//
//    }
    }
}

File: DessertPusher/app/src/main/java/com/salmakhd/dessertpusher/MainActivity.kt
package com.salmakhd.dessertpusher

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import timber.log.Timber

private const val TAG = "MainActivity"
class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        Log.i(TAG, "onCreate() callback called.")
        // alternative way
        Timber.i("onCreate() callback called (Timber-generated).")
    }

    override fun onStart() {
        super.onStart()
        Log.i(TAG, "onStart() callback called")
        Timber.i("onStart() callback called (Timber generated).")
    }

    override fun onResume() {
        super.onResume()
        Log.i(TAG, "onResume() callback called")
        Timber.i("onResume() callback called (Timber generated).")
    }

    override fun onPause() {
        super.onPause()
        Log.i(TAG, "onPause() callback called")
        Timber.i("onPause() callback called (Timber generated).")
    }

    override fun onStop() {
        super.onStop()
        Log.i(TAG, "onStop() callback called")
        Timber.i("onStop() callback called (Timber generated).")
    }
    override fun onRestart() {
        super.onRestart()
        Log.i(TAG, "onRestart() callback called")
        Timber.i("onRestart() callback called (Timber generated).")
    }

    override fun onDestroy() {
        super.onDestroy()
        Log.i(TAG, "onDestroy() callback called")
        Timber.i("onDestroy() callback called (Timber generated).")
    }
}

